[docker] support metadata labels

v1v opened the pull request:
It uses the metadata labels format explained in https://github.com/opencontainers/image-spec/blob/master/annotations.md?rgh-link-date=2021-07-08T07%3A26%3A24Z

slide commented on Dockerfile:
```diff
@@ -90,7 +90,7 @@ ARG BUILD_DATE
     org.label-schema.build-date=$BUILD_DATE \
     org.label-schema.schema-version="1.0" \
-    org.label-schema.vendor="Jenkins" \
+    org.label-schema.vendor="Jenkins project" \
```
Would we want "Jenkins Project" with a capital P?

v1v replied:
No strong opinion I used the one already defined in https://github.com/jenkinsci/docker-agent/pull/141/files#diff-77057272bfe150be8f21c133d8797affL33, so if capital P then I'll change the other PR

slide replied:
I don't have a strong opinion either and I just noticed that from docker-agent as well. We can always change it to capital P later if desired.

v1v commented:
🙏 Is there anything else to get this merged? Thanks

slide commented on Makefile:
```diff
@@ -9,28 +11,28 @@ shellcheck:
 
 build-debian:
-	docker build --file Dockerfile .
+	docker build --build-arg BUILD_DATE=$(BUILD_DATE) --file Dockerfile .
```
Just as an FYI, this would not be used until we move builds to our own infra. The Dockerhub builds don't allow setting build args from what I can see.

oleg-nenashev reviewed (changes requested):
We need to resolve the merge conflict :(

slide commented:
Also, just to reiterate, the BUILD_DATE will not be set unless we move Linux image building/publishing to our own infra from Dockerhub builds.

v1v commented:
I just merged and fixed the conflicts. In addition, I removed the BUILD_DATE https://github.com/jenkinsci/docker/pull/965/commits/db75ddc034b51923e43379da30403748b7dcf1ef

v1v commented on tests/test_helpers.bash:
```diff
@@ -73,6 +73,11 @@ function get_jenkins_password {
     docker logs "$SUT_CONTAINER" 2>&1 | grep -A 2 "Please use the following password to proceed to installation" | tail -n 1
 }
 
+function get_version {
+    local from="${DOCKERFILE:-Dockerfile}"
+    grep 'ENV JENKINS_VERSION' "${from}" | sed -e 's/.*:-\(.*\)}/\1/'
+}
+
```
I've just found I need to reimplement this particular function

v1v commented:
I think the failed tests are related to https://github.com/jenkinsci/docker/issues/1014

New tests are passing 
![image](https://user-images.githubusercontent.com/2871786/98270643-fa477d80-1f86-11eb-9410-8cb4533d48c2.png)

https://ci.jenkins.io/job/Packaging/job/docker/job/PR-965/492/testReport/C__Jenkins_workspace_Packaging_docker_PR-965_tests_runtime.Tests/ps1/Build___windows___Test____11_hotspot_windowsservercore_1809__test_jenkins_arguments_version_in_docker_metadata/

![image](https://user-images.githubusercontent.com/2871786/98270687-04697c00-1f87-11eb-9599-9c5096448536.png)

https://ci.jenkins.io/job/Packaging/job/docker/job/PR-965/492/testReport/jenkinsci.docker.debian/Default/Build___linux___slim___Test_slim____35__8_debian_buster_hotspot__test_version_in_docker_metadata/

v1v commented:
Is there anything else missing?

slide commented:
Looks like there are some conflicts that need to be resolved

timja commented:
@slide?

timja commented:
/ @oleg-nenashev

slide commented:
Just waiting on Oleg, he had requested changes.

garethjevans commented:
@oleg-nenashev are you ok with this?

slide commented:
One test failing on the windows side right now.

timja commented:
Looks like opencontainers is the replacement: https://github.com/opencontainers/image-spec/blob/master/annotations.md

timja commented:
@slide any chance you can check the windows failure?

timja reviewed (approved):
Lol the failure was do dumb, a new image was added and it just needed adding to.

🤕
